src/thread.rs

```diff
@@ -680,14 +785,35 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     }
 
     #[inline]
-    fn set_active_thread_name(&mut self, new_thread_name: Vec<u8>) {
+    fn set_thread_name(&mut self, thread: ThreadId, new_thread_name: Vec<u8>) {
         let this = self.eval_context_mut();
         if let Some(data_race) = &mut this.machine.data_race {
             if let Ok(string) = String::from_utf8(new_thread_name.clone()) {
-                data_race.thread_set_name(this.machine.threads.active_thread, string);
+                data_race.thread_set_name(thread, string);
             }
         }
-        this.machine.threads.set_thread_name(new_thread_name);
+        this.machine.threads.set_thread_name(thread, new_thread_name);
+    }
+
+    #[inline]
+    fn set_thread_name_wide(&mut self, thread: ThreadId, new_thread_name: Vec<u16>) {
+        let this = self.eval_context_mut();
+        if let Some(data_race) = &mut this.machine.data_race {
+            if let Ok(string) = String::from_utf16(&new_thread_name) {
+                data_race.thread_set_name(thread, string);
+            }
+        }
+        this.machine.threads.set_thread_name(
+            thread,
+            new_thread_name.into_iter().flat_map(u16::to_ne_bytes).collect(),
```

Comment:
`to_ne_bytes` is suspicious -- we never had to do this anywhere in Miri, so why do you need it here?

Reply:
Sorry about missing this! :/